fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/config/CPerception.java

```diff
@@ -0,0 +1,199 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.fed.application.config;
+
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.providers.SumoIndex;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.providers.TrafficLightIndex;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.providers.TrafficLightTree;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.providers.VehicleGrid;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.providers.VehicleIndex;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.providers.VehicleTree;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.providers.WallIndex;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.providers.WallTree;
+import org.eclipse.mosaic.lib.geo.GeoRectangle;
+import org.eclipse.mosaic.lib.gson.AbstractTypeTypeAdapter;
+import org.eclipse.mosaic.lib.util.gson.UnitFieldAdapter;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.annotations.JsonAdapter;
+import com.google.gson.reflect.TypeToken;
+
+import java.io.Serializable;
+
+/**
+ * Configuration for the perception backend used in the ApplicationSimulator
+ * to determine surrounding vehicles, traffic lights, and buildings.
+ */
+public class CPerception implements Serializable {
+
+    /**
+     * Backend for the spatial index providing vehicle information.
+     */
+    public CVehicleIndex vehicleIndex = new CVehicleIndex.Tree();
+
+    /**
+     * Backend for the spatial index providing traffic light information.
+     */
+    public CTrafficLightIndex trafficLightIndex = new CTrafficLightIndex();
+
+    /**
+     * Backend for the spatial index providing information about building walls.
+     */
+    public CWallIndex wallIndex = new CWallIndex();
+
+    /**
+     * Area defining the section of the map in which traffic objects (traffic lights, vehicles) should be held in the index.
+     * This is useful if only part of your network contains vehicles.
+     */
+    public GeoRectangle perceptionArea;
+
+    /**
+     * A base class for configuring the VehicleIndex implementation to use during perception. Based on a hidden "type" parameter,
+     * JSON deserialization chooses from Tree, Grid, or SUMO configuration. Possible type values are: "tree", "grid", "sumo"
+     */
+    @JsonAdapter(CVehicleIndexTypeAdapterFactory.class)
+    public static abstract class CVehicleIndex implements Serializable {
+
+        /**
+         * Defines if the vehicle index is enabled. Default: false
+         */
+        public boolean enabled = false;
```

Comment:
I had the same thought indeed. Even better is that the index is only updated when `getObjectsInRange` is called. The only issue I see is, that in all scenarios all traffic lights would be subscribed. I'm not sure right now if we really want this.. (see `UnitSimulator#registerTrafficLights()`)

Reply:
I decided to enable the vehicle index by default, since it does not add almost none overhead if the perception module is not used. Traffic light index requires many additional subscriptions, and wall index can't be used in most cases since we don't have scenarios with walls included, thus keeping them disabled by default.